src/com/google/enterprise/adaptor/filenet/ConfigOptions.java

```diff
@@ -96,26 +98,23 @@ public ConfigOptions(AdaptorContext context)
     logger.log(Level.CONFIG, "filenet.objectFactory: {0}", objectFactoryName);
 
     // TODO(jlacey): Replace this with a MessageFormat pattern.
-    String workplaceUrl = config.getValue("filenet.displayUrl");
-    if (workplaceUrl.endsWith("/getContent/")) {
-      workplaceUrl = workplaceUrl.substring(0, workplaceUrl.length() - 1);
-    }
-    if (workplaceUrl.contains("/getContent")
-        && workplaceUrl.endsWith("/getContent")) {
-      workplaceUrl += "?objectStoreName=" + objectStoreName
-          + "&objectType=document&versionStatus=1&vsId=";
-    } else {
-      workplaceUrl += "/getContent?objectStoreName=" + objectStoreName
-          + "&objectType=document&versionStatus=1&vsId=";
-    }
-    displayUrl = workplaceUrl;
-    logger.log(Level.CONFIG, "displayUrl: {0}", displayUrl);
+    String pattern = config.getValue("filenet.displayUrlPattern");
     try {
-      new ValidatedUri(displayUrl + percentEscape(Id.ZERO_ID.toString()))
+      URI uri = new URI(
+          getDisplayUrl(pattern, Id.ZERO_ID, Id.ZERO_ID, objectStoreName));
+      if (!uri.isAbsolute()) {
+        URI ceUri = new URI(contentEngineUrl);
+        pattern = ceUri.getScheme() + "://" + ceUri.getRawAuthority()
+            + (pattern.startsWith("/") ? "" : "/") + pattern;
+      }
+      new ValidatedUri(
+          getDisplayUrl(pattern, Id.ZERO_ID, Id.ZERO_ID, objectStoreName))
           .logUnreachableHost();
-    } catch (URISyntaxException e) {
+      this.displayUrlPattern = pattern;
+      logger.log(Level.CONFIG, "displayUrlPattern: {0}", pattern);
+    } catch (IllegalArgumentException | URISyntaxException e) {
       throw new InvalidConfigurationException(
-          "Invalid displayUrl: " + e.getMessage());
+          "Invalid displayUrlPattern: " + pattern, e);
```

Comment:
I would rather log before throwing exception, but it is just not feasible in this case.